torch/distributed/_pycute/layout.py

```diff
@@ -312,7 +320,9 @@ def left_inverse(layout):
 
 # Split a layout by the composition of B and the "rest"
 # Use tuples-of-layouts to perform this operation by-mode and None as no-op
-def logical_divide(layoutA, layoutB):
+def logical_divide(
+    layoutA: Layout, layoutB: Optional[Union[Layout, IntTuple, tuple[Any, ...]]]
```

Comment:
An alias for  Optional[Union[Layout, IntTuple, tuple[Any, ...]]] would be better; also the Any is too grabby